core/src/test/scala/unit/kafka/controller/ControllerRequestMergerTest.scala

```diff
@@ -285,6 +301,44 @@ class ControllerRequestMergerTest {
     assertEquals(toMap(expectedPartitions.asJava), toMap(liCombinedControlRequest.stopReplicaPartitionStates()))
   }
 
+  @Test
+  def testMergingDifferentStopReplicaPartitionStatesV0Request(): Unit = {
```

Comment:
The logic is quite similar to the testMergingDifferentStopReplicaPartitionStates method.
Can we extract the common logic into one shared method with the two parameters: 1) the IBP version and 2) shouldSetLeaderEpoch.

Reply:
Thanks, that is a really good suggestion. On it!